src/core/featurehistory.h

```diff
@@ -0,0 +1,124 @@
+#ifndef FEATUREHISTORY_H
+#define FEATUREHISTORY_H
+
+#include <QObject>
+#include <QTimer>
+#include <qgsproject.h>
+
+
+typedef QPair<QgsFeature, QgsFeature> OldNewFeaturePair;
+
+class FeatureHistory : public QObject
```

Comment:
@suricactus , big thanks for the documentation here, you're a role model to all of us :wink:

Reply:
:) just making my future life easier.